### ISSUE TITLE
pm.test("**** This should fail ****", false) returns pass

### ISSUE THREAD

**Issue by dbkaynor** [a85/POSTMan-Chrome-Extension#4340]
The following code samples return pass or fail when used in a test, however both return true.

pm.test("**** This should fail ****", false)
pm.test("**** This should pass ****", true)

**Comment by sivcan** [comment]
@dbkaynor - pm.test is actually using an implementation close to Chai and you can refer the code here [Chai JS Expect](https://github.com/chaijs/chai/blob/master/lib/chai/interface/expect.js).

Read #TIP 6 (BDD) - [API Testing tips from a postman professional](http://blog.getpostman.com/2017/07/28/api-testing-tips-from-a-postman-professional/)

The **test** expects an assertion error to be thrown for it to be failed else it'll be passed.
Even if you write - `pm.test('sample-test-pass', function () {});`
That'll pass.

But if you write  - `pm.test('sample-test-fail, function () { throw new Error(); });`
That'll fail.

And if you want to dig deeper, you may want to read more about [Chai JS](https://github.com/chaijs/chai/).

**Comment by cglacet** [comment]
I have a similar experience, for example:

```js
pm.test("a = b?", function() { 
    pm.expect("a").to.equal("b");
});
```

Does pass. 

The code you provided also passes: 

```js
pm.test('sample-test-fail', function () { throw new Error(); });
```

**Comment by cglacet** [comment]
Still no fix for this, I'm experiencing this in `8.3.0` and `8.3.1`, it seems there is no way to make any test fail (it used to work for me a while a go, like 2 years ago maybe).

**Comment by cglacet** [comment]
Ah, I think I found a way to make a request tests to crash, you need to put the query in a collection and then run the whole collection. But that's an horribly painful workaround to work with.
